utils.go

```diff
@@ -51,15 +76,15 @@ func JoinTraceFromHeader(
 			if err != nil {
 				return nil, err
 			}
-			contextIDMap[strings.TrimPrefix(key, ContextIDHTTPHeaderPrefix)] = unescaped
+			carrier.TracerState[strings.TrimPrefix(key, ContextIDHTTPHeaderPrefix)] = unescaped
```

Comment:
The HTTP Headers don't start with X-, is that expected?

Also these headers are not documented in the specification. Are `Open-Tracing-Context-Id-` and `Open-Tracing-Trace-Tags-` supposed to be used for HTTP communication with headers?